### DISCUSSION TITLE
Você usa o AdBlock (ou derivados)?

### DISCUSSION THREAD

**Discussion by YuriBrunetto** [frontendbr/forum#357] [Discussão]
Durante uns 2 anos eu usei o AdBlock e achava que era uma das maravilhas do universo. Só que, com o passar do tempo, percebi que talvez a coisa não era tão boa assim. Comecei a pensar com cabeça de desenvolvedor, de produtores de conteúdo que ganham a vida na internet e que o AdBlock bloqueia a renda do indivíduo. Muita gente usa o AdBlock até com um caráter mais inocente, apenas querendo evitar aqueles segundinhos antes de um vídeo do YouTube ou de anúncios chatos em websites.

[Após ouvir este podcast](http://www.b9.com.br/60805/podcasts/braincast/braincast-164-adblock-o-colapso-da-publicidade-online/), eu comecei a mudar de ideia. Acho que todos nós devíamos nos conscientizar disso e ajudar nossos amigos produtores de conteúdo a não perder seus respectivos "dinheiros"! 😄 

--

Acredito que o AdBlock funciona da seguinte maneira:
* o cara faz um array com várias palavrinhas-chave: `ad, ads, advertising, publicidade` etc, etc
* ele percorre todos os elementos do site para ver se tem algum com a `class` ou até mesmo `id` que fecha com algumas das palavras-chave do array acima, aplicando nela um `display: none` ou algo do gênero

O problema é que não sei como que isso funciona com o começo dos vídeos do YouTube... Mas enfim, tive uma ideia de criarmos um repositório com todas essas palavras-chave para que nós devs poderemos evitar que o AdBlock bloqueie os ads dos projetos que fizemos.

O que acham da ideia?

E se falei merda, por favor, desculpem uahuahau

Abraços, queridos!

**Comment by eduardojmatos** [comment]
Fala @YuriBrunetto ! Essa é uma excelente discussão! Aliás, outro ponto de vista, que confesso não ter escutado o podcast pra checar se falaram disso, é o da privacidade. A maioria desses anúncios são exibidos conforme o seu padrão de navegação. O AdBlock faz esse trabalho também de não executar determinados scripts que fazem esse tracking de profile.

A respeito de como produtores de conteúdo ganham a vida e de como a publicidade pode ser feita, vários podcasts, por exemplo, tem maneiras de resolver isso. O [MRG](https://jovemnerd.com.br/mrg/), por exemplo, por diversas vezes faz episódios de material, seja de quadrinhos, jogos ou livros, que são patrocinados por empresas. Não vejo de todo mal nisso, visto que eles fazem certa curadoria nesses conteúdos (pelo menos é o que eles falam e é o que acabo acreditando hehehe).

Eu [tenho usado o Brave com o DuckDuckGo](https://medium.com/nossa-coletividad/one-week-with-brave-and-duckduckgo-15ffee9128f0#.3zfiuntzv), justamente por conta dessa invasão de privacidade que sofremos ao visitar um simples blog ou fazer uma determinada pesquisa num buscador. Nós nos acostumamos tanto que até nem ligamos mais pra esse fato. Nosso padrão de navegação é vendido e usado pelas empresas - e sim, a maioria descreve isso nos termos de uso, que quase ninguém lê antes de usar um serviço (eu inclusive 😞 ).

Enfim, acrescento mais esse ponto na discussão do AdBlock! =D

**Comment by jonasporto** [comment]
O [Adblock](https://github.com/adblockplus) funciona basicamente, bloqueando requisições para urls que são identificadas como origem de ads(inclusive no caso de anuncios do youtube). Essa lista é enorme e é gerenciada nesse [site](https://forums.lanik.us/). 

Acredito que não tem uma forma fácil de evitar o adblock. Alguns sites exibem uma mensagem com instruções, solicitando ao usuário, colocar o site na whitelist. 

Vale uma análise de quantos dos seus usuários utilizam algum tipo de Adblocker, para pensar em uma abordagem melhor.

**Comment by karlprieb** [comment]
Queria deixar uma indicação para todos aqui :) [Brave browser](https://brave.com/). Ele vem com adblock e not track ativado por padrão. Só que o diferencial dele, para mim, não são estas features, mas sim uma alternativa a ads. Ao invés de criadores de conteúdos exibirem ads e arruinar performance, trackear usuários, etc. Eles podem receber uma renda direto do usuário com o Brave Payments. Deem um lida sobre isso, é bem legal :)

**Comment by DevCarlosAugusto** [comment]
Caras, eu amo o adBlock, sinceramente odeio propagandas intrusivas, mas em alguns sites onde a propaganda não é intrusiva eu libero no meu adblock para a whitelist. Não acho que o adblock seja um problema direto, mas sim o modo de se fazer propaganda. sempre que entro no facebook mesmo, tem propagandas na minha timeline que o adblock não bloqueia e como o @eduardojmatos citou, é uma invasão de privacidade, muitas vezes fiz uma pesquisa rápida no google e fica lá na minha timeline durante dias propagandas sobre produtos daquela pesquisa simples, deixando a timeline mais suja ainda....

Enfim, atualmente eu uso cada vez menos o chrome, tenho usado o ópera que tem um mecanismo de bloqueio de propagandas nas configurações dele, similar ao que o @karlprieb citou acima.

**Comment by ghost** [comment]
Uso e sempre usarei.

**Comment by YuriBrunetto** [comment]
@eduardojmatos legal cara!! muito interessante o que tu disse. vou ter que dar uma pesquisada mais a fundo nisso, claramente tem muita coisa que eu ainda não sei 😬  

## 

@jonasporto entendi! faz sentido mesmo. valeu!

**Comment by ranosi** [comment]
Cara eu uso com frequência também. Desativo em alguns poucos sites que a propaganda não seja intrusiva. Esse é o maior problema com algumas propagandas. Depois que passei a usar ele e tive que usar computadores onde não tinha instalado que fui ver a quantidade de coisa que aparece em alguns sites, é muito absurdo.

**Comment by barbier** [comment]
Desde que o AdBlock foi vendido eu dei preferência para começar a usar o uBlock.
Eu desbloqueio no youtube justamente pelos canais que eu assisto, embora os mesmo já tenham outras formas de se manter, como fazer vídeos patrocinados ou Patreon.
Sobre a forma dos produtores de conteúdo se manterem.... O Patreon se tornou uma saída, bem prática, acredito. Há anos eu também vejo links para paypal em blogs de devs, então doações diretas não algo novo.

**Comment by nicklima** [comment]
@jonasporto estranho isso de bloquear com base na URL. Já vi o AdBlock bloquear um anuncio de um site meu que tinha link interno. Acho que ele pega por palavras chaves como o @YuriBrunetto falou. 

Arrisco dizer que ele ainda deve identificar o formato das mídias, a classe do elemento que engloba essas mídias.

Digo isso porque no exemplo que citei, uso uma imagem com 720x90 dentro de uma div com class ADS. Achei um script que identifica se o usuário tem o ADBLOCK instalado pois quero exibir um conteúdo alternativo para esses visitantes.

Como criador de um site onde o retorno pelo conteúdo gerado é todo via ADSENSE acho essa discussão super interessante :smile:

**Comment by jonasporto** [comment]
@nicklima é possível que ele bloqueie também por palavra chave, mas quando eu olhei o código no github e acompanhei o fórum, pude confirmar que eles têm uma grande blacklist de urls(vinculadas a ads), que é frequentemente atualizada.

Com base nessa lista, eles conseguem bloquear Ads pelo [browser](https://developer.chrome.com/extensions/webRequest)(solicitando essa permissão na extensão do browser), com o mínimo de código possível.

Ex: Enquanto navego pela home da netshoes, o console me informa que o recurso foi bloqueado pelo cliente.

![adblocker](https://cloud.githubusercontent.com/assets/2723477/20248815/ea5aa6be-a9d2-11e6-91af-cc411b96210a.png)

**Comment by nicklima** [comment]
Massa @jonasporto! Não quis dizer que ele não fazia o bloqueio por URL, apenas que não deveria ser apenas isso :smile: 

Devem ser uma mistura de muito do que falamos no tópico. Eu estou adorando esse tópico porque pode sair muita coisa legal daqui. Estou tentando ver uma forma de usar um script para exibir um alerta (não um alert) solicitando que permita os ads. E vendo também a possibilidade de substituir o bloco do banner por outro caso o user tenha adblock. Ainda vendo como farei.  

Abraço

**Comment by tcelestino** [comment]
Deixei de usar o Adblock e passei só usar o Ghostery, que pra mim tem um "fluxo" de uso mais interessante e menos arbitario do que o Adblock. 

Com o Ghostery consigo gerenciar bem o que pode ser exibido, já que em alguns sites deixo recursos de publicidade visiveis. 

Mas para a discussão sobre o uso ou não do Adblock e seus derivados, o que mais vejo é que existem abusos, principalmente de portais de conteudo em relação a exibição de publicidade. Infelizmente é um custo que existe e cabe o mercado se adaptar as novas formas de consumo. Parece ser dificil, vide a discussão que sempre ocorre pela a Internet a fora, porém se as empresas que vivem de publicidade não criarem um modelo de negócio que realmente faça o usuário consumir publicidade, não vai adiantar de nada e cada vez mais usuários iram optar pelo o bloqueio.
